draft-ietf-quic-http.md

```diff
@@ -180,103 +180,231 @@ frames, exposing the data contained within as a reliable byte stream to the
-sending the request headers if there is no body.
+the stream management. An HTTP request/response consumes a pair of streams in
+each direction.
+
+Aside from the connection control stream, the beginning of each stream starts
+with a short stream header.  This stream header is used to identify the type of
+stream and to link the stream to another stream as necessary.  See
+{{stream-header}} for more details on the stream header.
+
+\[Editor's Note: the following is clearly busted.  Requests and responses can't
+reliably be cancelled as a result of this change.  We will need QPACK/QCRAM for
+this to work.] Request, response and push streams contain HPACK data which
+manipulates connection-level state.  Therefore, these streams MUST NOT be closed
+with a stream-level error.
+
+Streams must be opened sequentially, with no gaps.  Data streams SHOULD be
+opened after the request, response or push stream is opened.  Data streams are
+closed after transferring the body.  Data streams are not opened for messages
+that contain no body and messages that contain an empty body.
 
 HTTP does not need to do any separate multiplexing when using QUIC - data sent
 over a QUIC stream always maps to a particular HTTP transaction. Requests and
-responses are considered complete when the corresponding QUIC streams are closed
-in the appropriate direction.
+responses are considered complete when the corresponding QUIC streams are
+closed.
+
+
+## Stream Header
+
+The stream header is used to correlate requests and responses, including the
+response to server push with the PUSH_PROMISE containing the request.
+
+The stream header consists of a single octet that identifies the type of stream,
+plus any parameters that are specific to the stream type.
+
+
+## Stream Types
+
+The stream types are summarized in {{stream-type-table}}.
+
+| Stream Type        | Code | Section             |
+|:-------------------|-----:|:--------------------|
+| Connection Control | n/a  | {{stream-control}}  |
+| Request            | 0x00 | {{stream-request}}  |
+| Response           | 0x01 | {{stream-response}} |
+| Data               | 0x02 | {{stream-data}}     |
+| Push               | 0x03 | {{stream-push}}     |
+{: #stream-type-table title="Stream Types"}
+
+An endpoint MUST terminate the connection with an HTTP_INVALID_STREAM_HEADER
+error if it receives a stream header that uses an unknown or unsupported type.
+
+
+### Stream 1: Connection Control Stream {#stream-control}
+
+Stream 1 is opened by both client and server and is used for the SETTINGS frame
+immediately after the connection opens.  After the SETTINGS frame has been sent,
+this stream is used for PRIORITY and CANCEL_REQUEST frames.
+
+There is no stream header for a connection control stream; the stream is
+identified by its stream number.  The connection control stream contains a
+sequence of frames, starting with the SETTINGS frame.
+
+If an endpoint receives a stream type of 0xff on any stream other than stream 1,
```

Comment:
I think two versions collided here.  Stream type 0xff is undefined, so it's also invalid on Stream 1.